utils/validation.js

```diff
@@ -0,0 +1,261 @@
+// File: validation.js
+// Description: validates a submitted form against a list of constraints
+
+const NAME_MIN = 5;
+exports.NAME_MIN = NAME_MIN;
+const NAME_MAX = 50;
+exports.NAME_MAX = NAME_MAX;
+const CREDITS_MIN = 32;
+exports.CREDITS_MIN = CREDITS_MIN;
+const pool = require("./mysqlPool").pool;
+
+// checks that the submitted form data does not violate any constraints
+// returns a value that can be used to identify which constraint was violated
+function enforceConstraints(userId, planName, courses) {
+
+  return userConstraint(userId, planName, courses)
+    .then((conData) => {
+      return studentConstraint(conData[0], conData[1], conData[2]);
+    })
+    .then((conData) => {
+      return planNameConstraint(conData[0], conData[1], conData[2]);
+    })
+    .then((conData) => {
+      return zeroCourseConstraint(conData[0], conData[1], conData[2]);
+    })
+    .then((conData) => {
+      return duplicateCourseConstraint(conData[0], conData[1], conData[2]);
+    })
+    .then((conData) => {
+      return courseConstraint(conData[0], conData[1], conData[2]);
+    })
+    .then((conData) => {
+      return restrictionConstraint(conData[0], conData[1], conData[2]);
+    })
+    .then((conData) => {
+      return creditConstraint(conData[0], conData[1], conData[2]);
+    })
+    .then(() => {
+      console.log("Plan does not violate any constraints");
+      return 0;
+    })
+    .catch((conData) => {
+      // check if the error was from a constraint violation
+      if (conData[4]) {
+        return conData[4];
+      } else {
+        console.log("Error while trying to check constraints");
+        throw Error(conData[3]);
+      }
+
+    });
+
+}
+exports.enforceConstraints = enforceConstraints;
+
+// checks that the user exists
+function userConstraint(userId, planName, courses) {
+
+  return new Promise((resolve, reject) => {
+
+    const sql = "SELECT * FROM User WHERE userId=?;";
+    pool.query(sql, userId, (err, results) => {
+
+      if (err) {
+        console.log("Error checking user constraint");
+        reject([userId, planName, courses, err, 0]);
+      } else {
+
+        if (!results.length)
+          reject([userId, planName, courses, "", 1]);
+        else
+          resolve([userId, planName, courses, "", 0]);
+
+      }
+    });
+
+  });
+
+}
+
+// checks that the user is a student
+function studentConstraint(userId, planName, courses) {
+
+  return new Promise((resolve, reject) => {
+
+    const sql = "SELECT * FROM User WHERE userId=? AND role=0;";
+    pool.query(sql, userId, (err, results) => {
+
+      if (err) {
+        console.log("Error checking student constraint");
+        reject([userId, planName, courses, err, 0]);
+      } else {
+
+        if (!results.length)
```

Comment:
I think that single-statement conditionals should have curly braces, but the linter says otherwise. Discussion started on Discord.